app/controllers/course/survey/responses_controller.rb

```diff
@@ -37,11 +37,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
+      @response.unsubmit if can?(:manage, @survey)
+
+      if @response.update_attributes(response_unsubmit_params)
```

Comment:
Since we don't intend for users to unsubmit and update fields at the same time, we can just save the response:

```
if @response.save
```